### DISCUSSION TITLE
Is it possible use this midi sampler with i2s pcm5102a

### DISCUSSION THREAD

**Discussion by ograndebe** [marcel-licence/esp32_midi_sampler#93] [Q&A] [answered]
I Have an i2s pcm5102a DAC board here, I tried to read the code and see if it's possible to use this sampler with the DAC I Have, but I didn't find an answer.

Thank you

**Comment by marcel-licence** [comment, marked as answer]
Hi,

yes it is possible. All projects do support this DAC

**Comment by ograndebe** [comment]
Good. There's some guide teaching how to wire and configure the headers?
It's not clear to me the proper configuration and wiring.

**Reply by ograndebe** [reply]
Hi. I found this markdown file that explains how to wire and make it work https://github.com/marcel-licence/ML_SynthTools/blob/main/extras/ml_pcm5102a.md .
I found an issue, when I define OUTPUT_SAW_TEST the https://github.com/marcel-licence/esp32_midi_sampler breaks the build. (I'll open an issue at the proper repository)

So, after wiring properly, and enabling NOTE_ON_AFTER_SETUP I had no sound (using esp32_midi_sampler). After that, I tried the https://github.com/marcel-licence/esp32_basic_synth I hear the sound. 

I have many questions now, but I'll put the question in the proper places, Thank you in advance. 

PS:  I Believe that the build problem is related to lib version, I suggest that you start to create tags to manage library versions and in each example repository define the version requirement (my background is programming not electronics)

**Reply by marcel-licence** [reply]
Hi, I would agree having tags might help but it is very hard to test the different possible configurations. I decided to create a central lib to be able to fix problems for all projects at the same time.
I've seen the question about the opto-coupler. It might be deleted?

I started to create some markdown but it is not all linked. I will try to get a better documentation in future for easier navigation to different topics.

Regarding the wiring; I've added some output visible in the serial monitor on startup.

Thank you very much for your feedback.
